src/structures/CommandInteractionOptionResolver.js

```diff
@@ -239,6 +239,19 @@ class CommandInteractionOptionResolver {
     const option = this._getTypedOption(name, '_MESSAGE', ['message'], required);
     return option?.message ?? null;
   }
+
+  /**
+   * Gets the focused option.
+   * @param {boolean} [required=false] Whether to throw an error if the option is not found
+   * @returns {?(string|number)} The value of the option, or null if not set and not required
+   */
+  getFocused(required = false) {
```

Comment:
This is kinda annoying to deal with in typings right now. This function exists on normal command interactions, where focussed options will never exist, but when the interaction type is autocomplete then its always present, so passing true to this function feels wrong (you can't get an autocomplete interaction without a focused option). Maybe this also needs to be part of the `AutocompleteInteraction`? Thoughts?

Reply:
What I'm saying is the required option is a hack for typings purposes here. There's no real reason to actually use it. And yes that applies to context menu interactions too! We should probably be using type generics instead

Reply:
I guess since this is the only one that would always be required in an Autocomplete interaction it could be removed without necessarily creating an inconsistency but it should definitely be kept on the option resolver and throw an error when used on other interaction types